pyext/test.py

```diff
@@ -0,0 +1,299 @@
+
+attr.id = pysairedis.SAI_ROUTER_INTERFACE_ATTR_TYPE
+attr.value.s32 = pysairedis.SAI_ROUTER_INTERFACE_TYPE_PORT
+pysairedis.sai_attribute_t_arr_setitem(attrs, 1, attr)
+
+attr.id = pysairedis.SAI_ROUTER_INTERFACE_ATTR_PORT_ID
+attr.value.oid = portlist[0]
+pysairedis.sai_attribute_t_arr_setitem(attrs, 2, attr)
+
+status = router_interface_api.create_router_interface(poid, swid, 3, attrs)
+print "create router interface: " + str(status)
+
+rifid = pysairedis.sai_object_id_t_p_value(poid)
+
+pysairedis.delete_sai_object_id_t_p(poid)
+pysairedis.delete_sai_attribute_t_arr(attrs)
+
+print "rifid: " + hex(rifid)
+
+next_hop_api = pysairedis.sai_next_hop_api_t()
+
+status = pysairedis.sai_get_next_hop_api(next_hop_api)
+print "sai_get_next_hop_api: " + str(status)
+
+poid = pysairedis.new_sai_object_id_t_p()
+attrs = pysairedis.new_sai_attribute_t_arr(3)
+
+attr.id = pysairedis.SAI_NEXT_HOP_ATTR_TYPE
+attr.value.s32 = pysairedis.SAI_NEXT_HOP_TYPE_IP
+pysairedis.sai_attribute_t_arr_setitem(attrs, 0, attr)
+
+attr.id = pysairedis.SAI_NEXT_HOP_ATTR_IP
+attr.value.ipaddr = pysairedis.sai_ip_address_t_from_string("10.0.0.1")
+pysairedis.sai_attribute_t_arr_setitem(attrs, 1, attr)
+
+attr.id = pysairedis.SAI_NEXT_HOP_ATTR_ROUTER_INTERFACE_ID
+attr.value.oid = rifid
+pysairedis.sai_attribute_t_arr_setitem(attrs, 2, attr)
+
+status = next_hop_api.create_next_hop(poid, swid, 3, attrs)
+print "create next hop: " + str(status)
+
+nexthopid = pysairedis.sai_object_id_t_p_value(poid)
+
+pysairedis.delete_sai_object_id_t_p(poid)
+pysairedis.delete_sai_attribute_t_arr(attrs)
+
+print "nexthopid: " + hex(nexthopid)
+
+route_api = pysairedis.sai_route_api_t()
+
+status = pysairedis.sai_get_route_api(route_api)
+print "sai_get_route_api: " + str(status)
+
+attrs = pysairedis.new_sai_attribute_t_arr(1)
+
+attr.id = pysairedis.SAI_ROUTE_ENTRY_ATTR_PACKET_ACTION
+attr.value.s32 = pysairedis.SAI_PACKET_ACTION_DROP
+pysairedis.sai_attribute_t_arr_setitem(attrs, 0, attr)
+
+re = pysairedis.sai_route_entry_t()
+
+re.destination = pysairedis.sai_ip_prefix_t_from_string("0.0.0.0/0")
+re.switch_id = swid
+re.vr_id = vrid
+
+status = route_api.create_route_entry(re, 1, attrs)
+print "create default route entry: " + str(status)
+
+pysairedis.delete_sai_attribute_t_arr(attrs)
+
+attrs = pysairedis.new_sai_attribute_t_arr(1)
+
+attr.id = pysairedis.SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID
+attr.value.oid = nexthopid
+pysairedis.sai_attribute_t_arr_setitem(attrs, 0, attr)
+
+re = pysairedis.sai_route_entry_t()
+
+re.destination = pysairedis.sai_ip_prefix_t_from_string("100.1.0.1/32")
+re.switch_id = swid
+re.vr_id = vrid
+
+status = route_api.create_route_entry(re, 1, attrs)
+print "create route entry: " + str(status)
+
+pysairedis.delete_sai_attribute_t_arr(attrs)
+
+vlan_api = pysairedis.sai_vlan_api_t()
+
+status = pysairedis.sai_get_vlan_api(vlan_api)
+print "sai_get_vlan_api: " + str(status)
+
+poid = pysairedis.new_sai_object_id_t_p()
```

Comment:
where is this get implemented?

Reply:
which get ? sai_get_vlan_api? - in pysairedis.cpp file, this is custom helper function